src/libcollections/str.rs

```diff
@@ -1924,4 +1905,15 @@ impl str {
     pub fn escape_unicode(&self) -> String {
         self.chars().flat_map(|c| c.escape_unicode()).collect()
     }
+
+    /// Converts the `Box<str>` into a `String` without copying or allocating.
+    #[unstable(feature = "box_str",
+               reason = "recently added, matches RFC")]
+    pub fn into_string(self: Box<str>) -> String {
+        let len = self.len();
+
+        unsafe {
+            String::from_raw_parts(Box::into_raw(self) as *mut u8, len, len)
+        }
```

Comment:
This may be best implemented compositionally with a transmute, `<[_]>::into_vec` plus `String::from_utf8_unchecked`, that way there's no need to worry about the raw parts being exactly what we want, and there's exactly one point of `Box<[T]>` to `Vec<T>` (`into_vec`)